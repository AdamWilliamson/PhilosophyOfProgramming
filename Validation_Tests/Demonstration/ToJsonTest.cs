using ApprovalTests;
using System.Collections.Generic;
using Validations;
using Validations.Utilities;
using Xunit;

namespace Validations_Tests.Demonstration
{
    public class ToJsonTest
    {
        [Fact]
        public void BasicValidator_Describe_ToJson()
        {
            // Arrange
            var validator = new BasicValidator();
            var runner = new ValidationRunner<BasicValidatableObject>(new List<IValidator<BasicValidatableObject>>() { validator });

            // Act
            var results = runner.Describe();
            var json = new ErrorMessagePrettifier().ToJson(results);

			// Assert
			Approvals.VerifyJson(json);
        }

		[Fact]
		public void BasicValidator_Validate_ToJson()
		{
			// Arrange
			var validator = new BasicValidator();
			var runner = new ValidationRunner<BasicValidatableObject>(new List<IValidator<BasicValidatableObject>>() { validator });

			// Act
			var results = runner.Validate(new BasicValidatableObject());
			var json = new ErrorMessagePrettifier().ToJson(results);

			// Assert
			Approvals.VerifyJson(json);
		}
	}
}
