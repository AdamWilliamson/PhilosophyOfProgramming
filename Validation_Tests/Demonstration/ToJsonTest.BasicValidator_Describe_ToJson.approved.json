{
    "scopeMessage": null,
    "validator": null,
    "error": null,
    "children": [
        {
            "scopeMessage": null,
            "validator": null,
            "error": "Integer",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": "Equal To",
                    "error": "Must equal to 1",
                    "children": []
                },
                {
                    "scopeMessage": null,
                    "validator": "Equal To",
                    "error": "Must equal to 2",
                    "children": []
                }
            ]
        },
        {
            "scopeMessage": null,
            "validator": null,
            "error": "String",
            "children": [
                {
                    "scopeMessage": null,
                    "validator": "Equal To",
                    "error": "Must equal to ",
                    "children": []
                }
            ]
        }
    ]
}